sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/StorageCommonTestUtils.java

```diff
@@ -288,4 +300,49 @@ public static File getRandomFile(int size, TestResourceNamer testResourceNamer)
             throw new UncheckedIOException(e);
         }
     }
+
+    /**
+     * Gets token credentials for a test.
+     *
+     * @param interceptorManager The interceptor manager to use.
+     * @return The TokenCredential to use.
+     */
+    public static TokenCredential getTokenCredential(InterceptorManager interceptorManager) {
+        if (interceptorManager.isPlaybackMode()) {
+            return new MockTokenCredential();
+        } else if (interceptorManager.isRecordMode()){
+            return new DefaultAzureCredentialBuilder().build();
+        } else { //live
+            Configuration config = Configuration.getGlobalConfiguration();
+
+            ChainedTokenCredentialBuilder builder = new ChainedTokenCredentialBuilder()
+                .addLast(new EnvironmentCredentialBuilder().build())
+                .addLast(new AzureCliCredentialBuilder().build())
+                .addLast(new AzureDeveloperCliCredentialBuilder().build());
+
+            String serviceConnectionId = config.get("AZURESUBSCRIPTION_SERVICE_CONNECTION_ID");
+            String clientId = config.get("AZURESUBSCRIPTION_CLIENT_ID");
+            String tenantId = config.get("AZURESUBSCRIPTION_TENANT_ID");
+            String systemAccessToken = config.get("SYSTEM_ACCESSTOKEN");
```

Comment:
What is this?

Reply:
This is how we authenticate using AzurePipelinesCredential

Reply:
`SYSTEM_ACCESSTOKEN` is not an Entra ID app secret, correct?